src/SmartComponents/Drift/utils.js

```diff
@@ -0,0 +1,76 @@
+export function formatDate(date) {
+    const d = new Date(date);
+    return new Date(d.getTime() - d.getTimezoneOffset() * 60 * 1000).toISOString().split('T')[0];
+}
+
+;
+
+export function getDate(ago) {
+    let d = new Date();
+    d.setDate(d.getDate() - ago);
+    return formatDate(d);
+}
+
+export function parseDriftData(data) {
+    const payload = {};
+    payload.numEvents = data.meta.count;
+    payload.baselineDetail = [];
+    let unique_baselines = [];
+
+    data.data.forEach((item) => groupPayload(item.payload, payload.baselineDetail, unique_baselines));
+    payload.baselineDetail.sort(function (a, b) {
+        if (a.systems.length < b.systems.length) {
+            return 1;
+        }
+
+        if (a.systems.length > b.systems.length) {
+            return -1;
+        }
+
+        return 0;
+    });
+
+    return payload;
+}
+
+function groupPayload(data, array, unique_baselines) {
+    const payload = JSON.parse(formatString(data));
+    const host_id = payload.context.inventory_id;
+
+    let baseline_id = '';
+    let baseline_name = '';
+    for (let i = 0; i < payload.events.length; i++) {
+
+        baseline_id = payload.events[i].payload.baseline_id;
+        baseline_name = payload.events[i].payload.baseline_name;
+
+        //Validate if baselineId already exists
+        // If exists, validate if the current system exists
+        // else create baseline and system
+        if (unique_baselines.includes(baseline_id)) {
+            // Validate if system is associated with this baseline
+            array.forEach((item) => {
+                if (item.baselineId === baseline_id) {
+                    if (!item.systems.includes(host_id)) {
+                        item.systems.push(host_id);
+                    }
+                }
+            });
+        }
+        else {
+            // Add baseline data structure to the array
+            unique_baselines.push(baseline_id);
+
+            let newPayload = {
+                baselineId: baseline_id,
+                baselineName: baseline_name,
+                systems: [host_id]
+            };
+            array.push(newPayload);
+        }
+    }
+}
+
+function formatString(item) {
+    return item.replaceAll('\\', '').replaceAll('"{', '{').replaceAll('}"', '}');
```

Comment:
This is not a good way of formatting escaped JSON. Much better would be to call `JSON.parse` twice
```suggestion
    return JSON.parse(JSON.parse(item));
```

Reply:
This isn't working. Keep the old way

Reply:
We shouldn't send non parsable data over the REST api. You should definetely raise a request with yout API team to send correct data. String with escaped JSON which is not parsable is not a correct way.

Reply:
I agree, I will open a ticket on notifcations side